### ISSUE TITLE
deseq2 bug fix

### ISSUE THREAD

**Issue by gungne** [galaxyproject/tools-iuc#1221]
deseq2 is not loading correctly either on cloud(16.07/16.10) or local instance(17.01).
So adding the additional requirement will fix the problem:

>         <requirement type="package">r-getopt</requirement>
>         <requirement type="package">r-rjson</requirement>
>         <requirement type="package">r-gplots</requirement>
 

cheers,
Jacky

**Comment by nsoranzo** [comment]
All 3 requirements you mentioned are already dependencies of the Conda package, which is the only dependency resolver the tool supports now. Try to install (or uninstall and reinstall) the latest release of the `deseq2` repository from the Tool Shed on a Conda-enabled Galaxy (17.01 has Conda enabled by default).
If this does not work, please post the Galaxy logs produced while installing the repository and/or running the tool.

**Comment by gungne** [comment]
Hi Nicola,
Thanks for the reply.
Yes and no, the conda is doing great job in downloading these three packages. However, I still encounter the missing package problem in actual running the deseq2. 
So basically, my idea for the bug is that: the conda install the dependencies but galaxy unable to use them(dependencies in the galaxy folder). When I installed the three packages in the R.framework or add the requirement in the xml file, deseq2 works perfectly.

Here, thanks for Mo Heydarian`s help. I reproduce the exact error and I save all the logs. Hope they help. 

My workflow:
1/Manually delete dependencies in R.framework
Run Rstudio to check: 
Rstudio commendline:
> c("rjson")  %in% installed.packages()[,1]
[1] FALSE
> c("gplots")  %in% installed.packages()[,1]
[1] FALSE
> c("getopt")  %in% installed.packages()[,1]
[1] FALSE

2/Manually delete and reinstall galaxy
Jis-MacBook-Air:~ Gungnir$ git clone -b release_17.01 https://github.com/galaxyproject/galaxy.git
Cloning into 'galaxy'...

remote: Counting objects: 263683, done.
remote: Compressing objects: 100% (10/10), done.
remote: Total 263683 (delta 0), reused 0 (delta 0), pack-reused 263673
Receiving objects: 100% (263683/263683), 196.46 MiB | 7.72 MiB/s, done.
Resolving deltas: 100% (209157/209157), done.
Checking out files: 100% (4416/4416), done.

3/Run galaxy
galaxy_installation.log

4/Run instance
galaxy_startup.log

5/Register
create_user.log

6/Edit galaxy.ini.sample / save as galaxy.ini

7/Run again 
galaxy_startup2.log

8/Install deseq2
deseq2_installation.log

9/Import data and run
data_import_deseq2.log


[Deseq2 debug logs.zip](https://github.com/galaxyproject/tools-iuc/files/886877/Deseq2.debug.logs.zip)

**Comment by nsoranzo** [comment]
@gungne Inside the `deseq2_installation.log` file you attached there a bunch of error messages that start with:
```
PaddingError: Placeholder of length '80' too short in package
```

I hope this will be fixed by #1231 where I've updated the `deseq2` requirement to the latest Bioconda package.
